processing/src/main/java/org/apache/druid/frame/field/ComplexFieldReader.java

```diff
@@ -89,17 +89,43 @@ public boolean isNull(Memory memory, long position)
     return memory.getByte(position) == ComplexFieldWriter.NULL_BYTE;
   }
 
-  @Override
-  public boolean isComparable()
+  /**
+   * Alternative interface to read the field from the byte array without creating a selector and field pointer. It is much
+   * faster than wrapping the byte array in Memory for reading.
+   */
+  @Nullable
+  public static Object readFieldFromByteArray(
+      final ComplexMetricSerde serde,
+      final byte[] bytes,
+      final int position
+  )
   {
-    return true;
+    final byte nullByte = bytes[position];
+
+    if (nullByte == ComplexFieldWriter.NULL_BYTE) {
+      return null;
+    } else if (nullByte == ComplexFieldWriter.NOT_NULL_BYTE) {
+      // Reads length in big-endian format
```

Comment:
I think this is actually little-endian format? The least significant byte is in the first position.